src/jit-diff/jit-diff.cs

```diff
@@ -830,239 +835,330 @@ public static int InstallCommand(Config config)
             return ret;
         }
 
-        // Returns a count of the number of failures.
-        private static int RunDasmTool(List<string> commandArgs, List<string> assemblyPaths)
+        public class DiffTool
         {
-            int dasmFailures = 0;
-            List<string> allArgs = commandArgs;
-            foreach (var assemblyPath in assemblyPaths)
+            private class DasmWorkItem
             {
-                commandArgs.Add(assemblyPath);
-                Console.WriteLine("Diff command: {0} {1}", s_asmTool, String.Join(" ", allArgs));
-                CommandResult result = TryCommand(s_asmTool, allArgs);
-                commandArgs.Remove(commandArgs.Last());
-                if (result.ExitCode != 0)
+                public List<string> DasmArgs { get; set; }
+
+                public DasmWorkItem(List<string> dasmArgs)
                 {
-                    Console.Error.WriteLine("Dasm command returned with {0} failures", result.ExitCode);
-                    dasmFailures += result.ExitCode;
+                    DasmArgs = dasmArgs;
                 }
             }
-            return dasmFailures;
-        }
 
-        // Returns:
-        // 0 on success,
-        // -1 on configuration failure (e.g., JIT not found),
-        // Otherwise, a count of the number of failures generating asm, as reported by the asm tool.
-        private static int RunDasmTool(Config config, List<string> commandArgs, List<string> assemblyPaths, string tag, string clrPath)
-        {
-            List<string> dasmArgs = commandArgs.ToList();
-            dasmArgs.Add("--tag");
-            dasmArgs.Add(tag);
-            dasmArgs.Add("--crossgen");
-            if (config.HasCrossgenExe)
+            private Config m_config;
+            private ConcurrentQueue<DasmWorkItem> DasmWorkQueue = new ConcurrentQueue<DasmWorkItem>();
+
+            public DiffTool(Config config)
+            {
+                m_config = config;
+            }
+
+            // Returns a count of the number of failures.
+            private void RunDasmTool()
             {
-                dasmArgs.Add(config.CrossgenExe);
-                dasmArgs.Add("--jit");
+                int dasmFailures = 0;
 
-                var clrjitPath = FindJitLibrary(clrPath);
-                if (clrjitPath == null)
+                while (true)
                 {
-                    Console.Error.WriteLine("clrjit not found in " + clrPath);
-                    return -1;
+                    DasmWorkItem item;
+                    if (DasmWorkQueue.TryDequeue(out item))
```

Comment:
> (AFAIR it's possible to run just the diff part but I've never tried that myself so I don't know

You're recalling correctly, and that's what I've been doing, both to reconcile error counts (which is now moot) and because it lets me be lazy and always invoke jit-diff through a wrapper script that fills in all the paths for me (assuming the jit in bin\Product\blah-Checked is what I want) so I can just build+gen-diff+build-master+gen-base rather than keeping track of where I'm putting different jit dlls.

Which is just a long-winded way of saying that I for one would appreciate making things faster even if I don't run base/diff together, although some sort of --max-threads option to avoid hogging the CPU would be appreciated too.